provider/aws/aws.go

```diff
@@ -415,48 +415,86 @@ func (p *AWSProvider) submitChanges(ctx context.Context, changes []*route53.Chan
 
 	var failedZones []string
 	for z, cs := range changesByZone {
-		var failedUpdate bool
+		errors := p.applyWithBisect(ctx, cs, z, aws.StringValue(zones[z].Name))
+		if len(errors) > 0 {
+			failedZones = append(failedZones, z)
+		}
+	}
 
-		batchCs := batchChangeSet(cs, p.batchChangeSize)
+	if len(failedZones) > 0 {
+		return fmt.Errorf("failed to submit all changes for the following zones: %v", failedZones)
+	}
 
-		for i, b := range batchCs {
-			for _, c := range b {
-				log.Infof("Desired change: %s %s %s [Id: %s]", *c.Action, *c.ResourceRecordSet.Name, *c.ResourceRecordSet.Type, z)
-			}
+	return nil
+}
 
-			if !p.dryRun {
-				params := &route53.ChangeResourceRecordSetsInput{
-					HostedZoneId: aws.String(z),
-					ChangeBatch: &route53.ChangeBatch{
-						Changes: b,
-					},
-				}
+func (p *AWSProvider) applyWithBisect(ctx context.Context, changes []*route53.Change, z string, zoneName string) []error {
+	res := []error{}
 
-				if _, err := p.client.ChangeResourceRecordSetsWithContext(ctx, params); err != nil {
-					log.Errorf("Failure in zone %s [Id: %s]", aws.StringValue(zones[z].Name), z)
-					log.Error(err) //TODO(ideahitme): consider changing the interface in cases when this error might be a concern for other components
-					failedUpdate = true
-				} else {
-					// z is the R53 Hosted Zone ID already as aws.StringValue
-					log.Infof("%d record(s) in zone %s [Id: %s] were successfully updated", len(b), aws.StringValue(zones[z].Name), z)
-				}
+	batchCs := batchChangeSet(changes, p.batchChangeSize)
 
-				if i != len(batchCs)-1 {
-					time.Sleep(p.batchChangeInterval)
-				}
+	for i, b := range batchCs {
+		if p.ChangeZone(ctx, b, z, zoneName) {
+			changesByName, names := getChangesByName(b)
+
+			if len(names) == 1 {
+				return []error{fmt.Errorf("failed to submit changes: %v", b)}
 			}
+
+			size := len(names) / 2
```

Comment:
We might want to ensure the minimum size is 2 because the main record and the ownership TXT records should be applied in the same transaction so that they either are both or applied or not at all.

Reply:
For safety we should also ensure that `names` is even I guess. Otherwise this logic might become incorrect.